src/features/CartList/hooks/useControlQuantity.ts

```diff
@@ -0,0 +1,66 @@
+import { useContext } from 'react';
+
+import { addCartItem, deleteCartItem, getCartItemList, updateCartItem } from '@/api/cart';
+import { ToastContext } from '@/shared/context/ToastProvider';
+import { useData } from '@/shared/context/useData';
+
+export const useControlQuantity = (productId: number) => {
+  const { cartData } = useData();
+  const { showToast } = useContext(ToastContext);
+
+  const cartItem = cartData.data?.find((item) => item.product.id === productId);
+
+  const currentQuantity = cartItem?.quantity || 0;
+  const isInCart = !!cartItem;
+
+  const increaseQuantity = async () => {
+    try {
+      if (!isInCart) {
+        await cartData.mutate(
+          () => addCartItem({ productId: productId, quantity: 1 }),
+          getCartItemList
+        );
+      }
+      if (isInCart) {
+        await cartData.mutate(
+          () => updateCartItem({ cartId: cartItem.id, newQuantity: currentQuantity + 1 }),
+          getCartItemList
+        );
+      }
+    } catch (error) {
+      const errorResponse = (error as Error)?.message;
+      showToast(
+        errorResponse ??
+          `"${cartItem?.product.name}" 상품의 최대 구매 수량은 ${cartItem?.product.quantity}개 입니다.`
+      );
+    }
+  };
+
+  const removeCartItem = async () => {
+    try {
+      if (cartItem) {
+        await cartData.mutate(() => deleteCartItem(cartItem.id), getCartItemList);
+      }
```

Comment:
소소하게는 가드문을 위에서 먼저 사용하는 방법을 생각해 볼 수도 있겠네요

```suggestion
      if (!cartItem) return 
        
      await cartData.mutate(() => deleteCartItem(cartItem.id), getCartItemList);
```

Reply:
헙 수정했습니다! 감사합니다! 🫡 [refactor: removeCartItem 함수 보호 구문 적용](https://github.com/woowacourse/react-shopping-products/pull/134/commits/b655f516af587805ea049522a55d628716265944)